controllers/networkchaos/trafficcontrol/types.go

```diff
@@ -128,7 +128,10 @@ func (r *endpoint) Apply(ctx context.Context, req ctrl.Request, chaos v1alpha1.I
 			return err
 		}
 	case v1alpha1.Both:
-		err = r.applyTc(ctx, pods, pods, externalCidrs, m, networkchaos)
+		if len(targets) > 0 {
+			targets = pods
+		}
+		err = r.applyTc(ctx, pods, targets, externalCidrs, m, networkchaos)
```

Comment:
I have a question, I see that we resolve the "direction: both" with certain codes, it also bring another issue: https://github.com/chaos-mesh/chaos-mesh/issues/1450

Why not this:

```go
case v1alpha1.Both:
    err = r.applyTc(ctx, sources, targets, externalCidrs, m, networkchaos)
    // resolve error
    err = r.applyTc(ctx, targets, sources, []string{}, m, networkchaos)
    // resolve error
```

It means "direction: both" is the composition of both "direction: from" and "direction: to".

Reply:
Great! I will fix it.

Reply:
This has another problem :crying_cat_face: . #1448 assumes that the ipset with the same name should have the same content. However, if you `applyTc` twice, they will have the same `ipset` name with different content.

```
dstIpset := ipset.BuildIPSet(targets, externalTargets, networkchaos, string(tcType)[0:5], m.Source)
```

The `postFix` should be modified from `string(tcType)[0:5]` to the `tgt`, `src` like the partition chaos